src/hnsw/options.c

```diff
@@ -54,6 +60,17 @@ int HnswGetEf(Relation index)
     return HNSW_DEFAULT_EF;
 }
 
+char *HnswGetIndexFilePath(Relation index)
+{
+    HnswOptions *opts = (HnswOptions *)index->rd_options;
+    if(!opts) return NULL;
+    if(!opts->experimantal_index_path_offset) {
+        return NULL;
+    }
+
+    return (char *)opts + opts->experimantal_index_path_offset;
```

Comment:
Oh, this makes so much sense - variable length options are stored in the struct indirectly, via an offset.
This is what I had missed when I could not get string options working